### PR TITLE
Allow specifying error handling scheme for decoding bytes/json resp (SYN-4240)

### PR DESCRIPTION
**Pull request by Cisphyx**

### CODE REVIEW

File: synapse/lib/stormhttp.py [hunk 1]
```diff
@@ -448,9 +452,20 @@ def getObjLocals(self):
             'msgpack': self._httpRespMsgpack,
         }
 
-    async def _httpRespJson(self):
+    async def _httpRespJson(self, encoding=None, errors='surrogatepass'):
         try:
-            return json.loads(self.valu.get('body'))
+            valu = self.valu.get('body')
+
```
Comment (vEpiphyte) [inline synapse/lib/stormhttp.py:458]: errors should be tostr()`d

Comment (Cisphyx) [reply synapse/lib/stormhttp.py:459]: oops, good call

File: synapse/lib/stormhttp.py [hunk 2]
```diff
@@ -448,9 +452,21 @@ def getObjLocals(self):
             'msgpack': self._httpRespMsgpack,
         }
 
-    async def _httpRespJson(self):
+    async def _httpRespJson(self, encoding=None, errors='surrogatepass'):
         try:
-            return json.loads(self.valu.get('body'))
+            valu = self.valu.get('body')
+            errors = await s_stormtypes.tostr(errors)
```
Comment (vEpiphyte) [inline synapse/lib/stormhttp.py:458]: noneok=True

Comment (Cisphyx) [reply synapse/lib/stormhttp.py:458]: errors has to be a str though

Comment (vEpiphyte) [reply synapse/lib/stormhttp.py:458]: oh. was misreading.

**Review by vEpiphyte — APPROVED** [review]

**Pull request merged**